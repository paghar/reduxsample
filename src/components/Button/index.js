import React from "react"
import { Wrapper } from "./index.style"
const Button = ({ clicked, isStartTimer }) => {
  return (
    <Wrapper onClick={clicked}>
      <span>
        {isStartTimer ? "Start Random Rating" : "Stop Random Rating"}{" "}
      </span>
    </Wrapper>
  )
}

export default Button
