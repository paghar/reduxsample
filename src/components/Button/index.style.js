import styled from "styled-components"

export const Wrapper = styled.button`
  border-radius: 4px;
  background: linear-gradient(
    236deg,
    rgba(252, 201, 169, 1) 13%,
    rgba(255, 253, 252, 1) 28%,
    rgba(252, 201, 169, 1) 52%,
    rgba(255, 253, 252, 1) 78%,
    rgba(252, 201, 169, 1) 93%
  );
  border: 2px solid #ff66b3;
  color: #ff66b3;
  text-align: center;
  text-transform: uppercase;
  font-size: 16px;
  padding: 10px 30px;
  transition: all 0.4s;
  cursor: pointer;
  margin: 5px;
  -webkit-user-select: none;  
  -moz-user-select: none;    
  -ms-user-select: none;     
  user-select: none;         

  :hover {
    border: 2px solid #ff1a8c;
    color: #ff1a8c;
    -webkit-box-shadow: 0px 0px 26px -1px rgba(255, 102, 179, 0.45);
    box-shadow: 0px 0px 26px -1px rgba(255, 102, 179, 0.45);
  }
`
