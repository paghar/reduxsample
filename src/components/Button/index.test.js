import React from "react"
import renderer from "react-test-renderer"
import Button from "."
import { shallow } from "enzyme"
describe("<Button />", () => {
  it("Button", () => {
    const tree = renderer.create(<Button isStartTimer={false} />).toJSON()
    expect(tree).toMatchSnapshot()
  });

  it("Button Click Event", () => {
    const mockCallBack = jest.fn()
    const wrapper = shallow(
      <Button isStartTimer={false} clicked={mockCallBack} />
    )
    wrapper.simulate("click")
    expect(mockCallBack).toHaveBeenCalled()
  });
})
