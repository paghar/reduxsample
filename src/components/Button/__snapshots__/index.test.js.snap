// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> Button 1`] = `
<button
  className="sc-bdVaJa cpOyOP"
>
  <span>
    Stop Random Rating
     
  </span>
</button>
`;
